### PR TITLE
cdk2: add CHECK and DISCOVER operations

### PR DESCRIPTION
**Pull request by postamar**
This PR adds full-fledged CHECK and DISCOVER operation  with SSH tunnel support. At this point in the PR stack we actually start opening JDBC connections and running queries. This PR adds plenty of tests and test fixtures. We rely on H2 to serve as our test database for the fake source connector, it's good enough for what we need.

The SSH tunnel code was lifted with only superficial changes from the existing CDK. Same goes for the SSH bastion container.

### CODE REVIEW

File: airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/DiscoverOperation.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import io.airbyte.protocol.models.v0.AirbyteStream
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Singleton
+
+private val log = KotlinLogging.logger {}
+
+@Singleton
+@Requires(property = CONNECTOR_OPERATION, value = "discover")
+@Requires(env = ["source"])
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/DiscoverOperation.kt:23]: nice - we are still using these ENV tags

File: airbyte-integrations/connectors/source-oracle-v2/src/cdk/test/kotlin/io/airbyte/cdk/jdbc/H2TestFixtureTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+package io.airbyte.cdk.jdbc
+
+import java.sql.DriverManager
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+
+class H2TestFixtureTest {
+
+    val h2 = H2TestFixture()
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-oracle-v2/src/cdk/test/kotlin/io/airbyte/cdk/jdbc/H2TestFixtureTest.kt:13]: ... what's H2?

File: airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/DiscoverOperation.kt (hunk 1 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-oracle-v2/src/cdk/main/kotlin/io/airbyte/cdk/operation/DiscoverOperation.kt:23]: This here might not be the most relevant use case (it's probably redundant) but constraints on `env` are particularly useful in tests.

File: airbyte-integrations/connectors/source-oracle-v2/src/cdk/test/kotlin/io/airbyte/cdk/jdbc/H2TestFixtureTest.kt (hunk 2 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-oracle-v2/src/cdk/test/kotlin/io/airbyte/cdk/jdbc/H2TestFixtureTest.kt:13]: It's java's equivalent of sqlite: https://en.wikipedia.org/wiki/H2_(database)

**Pull request closed without merging**